src/bin/cargo/main.rs

```diff
@@ -168,7 +168,17 @@ fn execute_external_subcommand(config: &Config, cmd: &str, args: &[&str]) -> Cli
         None => {
             let suggestions = list_commands(config);
             let did_you_mean = closest_msg(cmd, suggestions.keys(), |c| c);
-            let err = anyhow::format_err!("no such subcommand: `{}`{}", cmd, did_you_mean);
+
+            let err = if cmd.contains('+') {
+                anyhow::format_err!("no such subcommand: `{}`{}\n\n\tCargo does not handle `+toolchain` directives itself.\n\tDid you mean to run `cargo` through `rustup` instead?", cmd, did_you_mean)
```

Comment:
I feel the wording can be improved here but not quite sure what it should be yet.